language/move-prover/stackless-bytecode-generator/src/borrow_analysis.rs

```diff
@@ -561,6 +590,34 @@ impl<'a> BorrowAnalysis<'a> {
         Ok(post)
     }
 
+    fn mark_dirty(&self, state: &mut BorrowState, ref_id: RefID) {
+        if let Some(node) = self.ref_id_to_borrow_node.get(&ref_id).cloned() {
+            // We could stop if the node is already in the set, but the graph can change as we
```

Comment:
I did not understand the comment.  What is the activity that can cause the "graph can change"?  If graph is referring to the borrow graph, then this graph does not appear to be changing while mark_dirty is executing.  I suspect you might be referring to changes to the graph during the dataflow analysis.

You are doing a DFS on an acyclic graph without maintaining a visited set.  Although not needed for termination, a visited  set might be more efficient.

Reply:
I extended the comment to make it clearer. The "activity" are multiple assignments to the same variable. I don't think it is worth the effort to add a visited set, because I expect borrow graphs have the size of just a few nodes.